File: automacao_api/features/objects/create.rb
class CriarRecurso
  include HTTParty
  format :json
  headers 'Content-Type' => 'application/json'

  def criar_sucesso
    @user_id = 1
    body = {
      "title": ('A'..'Z').to_a.shuffle[0,5].join,
      "body":  ('A'..'Z').to_a.shuffle[0,13].join,
      "userId": @user_id}.to_json
    self.class.base_uri (CONFIG['jsonplaceholder-url']).to_s
    response = self.class.post('/posts', body: body)
    return [response, @user_id]
  end
 end

File: performance/Tests/create.py
from typing import Any, Callable

from locust import HttpLocust, TaskSet, task, between


class Create(TaskSet):

    @task
    def post_create(self):
        uri = "/posts"
        payload = {
        "title": "foo",
        "body": "bar",
        "userId": 1
        }
        headers = {'content-type': 'application/json'}

        resp = self.client.post(uri, json=payload, headers=headers)
        print(resp.text)


class FindCreate(HttpLocust):
    task_set = Create
    wait_time = between(1.0, 4.0)
    host = "https://jsonplaceholder.typicode.com"
